### PR TITLE
cmds/boot/localboot: Add systemdboot/boot_loader_specification parsing

### PR DESCRIPTION
**Pull request by CodingVoid**
systemd-boot is getting more popular and is even used by some Live ISO Images, therefore we need a way to pass the config files in order to find the linux kernel, initramfs and cmdline. systemd-boot config files are based on the boot loader specification:
https://web.archive.org/web/20230410233456/https://uapi-group.org/specifications/specs/boot_loader_specification/

### CODE REVIEW

**Comment by hugelgupf** [comment]
The boot command already has BLS support. Wouldn't it be nicer to merge
these rather than keep duplicating?

File: cmds/boot/localboot/systemdboot.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
```
Comment (rminnich) [inline cmds/boot/localboot/systemdboot.go:1]: I'd like to try to maintain the pretense that we can run under other kernels, which I realize we have not been good about. Anyway, at some point, this probably ought to be called 
systemdboot_linux.go

**Review by rminnich — APPROVED** [review]

**Comment by CodingVoid** [comment]
> The boot command already has BLS support. Wouldn't it be nicer to merge these rather than keep duplicating?
my bad. I didn't see u-root already has BLS support. I guess I will switch to using the boot cmd instead of the localboot cmd and abort this change.
By the way are there any plans on deprecating the localboot cmd or are there still features that are only implemented in the localboot cmd that need to be put into the boot cmd?

**Comment by rminnich** [comment]
I'm going to take your word that you want to drop this change ...

**Pull request closed without merging**